### PR TITLE
Docker container excessively privileged running root user

### PR DESCRIPTION
**Pull request by zhang-hua**
* Add default non-root user 'pcsuser'
* Install sudo and add 'pcsuser' as sudoer
* Run service as 'pcsuser'

PBI[2211778]

# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Without explicitly defining a running user in a Dockerfile definition, the root user is utilized by default.
Each respective Dockerfile should explicitly define a user and mandate that user to be the service runner via the USER instruction. Further, the service user should also have a defined GROUP. Without one, the primary group for the user will fall back to the root group.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: scripts/docker/Dockerfile [hunk 1]
```diff
@@ -4,6 +4,10 @@ MAINTAINER Devis Lucato (https://github.com/dluc)
 
 LABEL Tags="Azure,IoT,Solutions,React,SPA"
 
+ARG user=pcsuser
```
Comment (dluc) [inline scripts/docker/Dockerfile]:
```
ARG user=app
```

File: scripts/docker/Dockerfile [hunk 2]
```diff
@@ -4,6 +4,10 @@ MAINTAINER Devis Lucato (https://github.com/dluc)
...
+
+RUN adduser -D -h /home/$user -s /bin/bash $user $user
```
Comment (dluc) [inline scripts/docker/Dockerfile]:
```
RUN groupadd -r $user && useradd --no-log-init -r -g $user $user
```

File: scripts/docker/content/run.sh [hunk 3]
```diff
@@ -3,4 +3,4 @@ set -ex
 
 # serve the app via nginx
 mkdir -p /app/logs
-nginx -g 'daemon off;' -c /app/nginx.conf
```
Comment (dluc) [inline scripts/docker/content/run.sh]: what's this change for?

Comment (zhang-hua) [reply scripts/docker/content/run.sh]: this is moved into nginx.conf.

File: scripts/docker/Dockerfile (hunk 2 above)
Comment (zhang-hua) [reply scripts/docker/Dockerfile]: 'groupadd' and 'useradd' command are not supported in /bin/sh. I will change them to use 'addgroup' and 'adduser' which works in sh.

**Comment by ppathan** [comment]
@zhang-hua Have you tested this change with hosting your own docker image in vm?

File: scripts/docker/Dockerfile [hunk 4]
```diff
@@ -4,6 +4,10 @@ MAINTAINER Devis Lucato (https://github.com/dluc)
 
 LABEL Tags="Azure,IoT,Solutions,React,SPA"
 
+ARG user=app
+
+RUN addgroup $user && adduser -D -G $user $user
```
Comment (dluc) [inline scripts/docker/Dockerfile]: can you sync this with the line of the other PR?

Comment (zhang-hua) [reply scripts/docker/Dockerfile]: the command 'groupadd' and 'useradd' are not supported by /bin/sh in Busybox. They are supported by bash shell. I have to change them to 'addgroup' and 'adduser' which are supported by /bin/sh.

**Comment by zhang-hua** [comment]
@ppathan , I did test the PR with private built images before pushing this PR. please also help to validate it if you have time.

File: scripts/docker/Dockerfile (hunk 4 above)
Comment (zhang-hua) [reply scripts/docker/Dockerfile]: the command 'groupadd' and 'useradd' are not supported by /bin/sh in Busybox. They are supported by bash shell. I have to change them to 'addgroup' and 'adduser' which are supported by /bin/sh.

**Comment by ppathan** [comment]
@dluc Any reason we are using bin/sh and not bash for docker image of webui?

**Pull request merged**